Testnet/DAOContract/DAOContract/DAOContract.cs

```diff
@@ -0,0 +1,213 @@
+
+        SetVote(proposalId, Message.Sender, ToVote(vote));
+    }
+
+    private void SetVote(uint proposalId, Address address, Votes vote)
+    {
+        var currentVote = (Votes)GetVote(proposalId, address);
+
+        if (currentVote == vote)
+        {
+            return;
+        }
+
+        Unvote(proposalId, currentVote);
+        State.SetUInt32($"Vote:{proposalId}:{address}", (uint)vote);
+
+        if (vote == Votes.Yes)
+        {
+            SetYesVotes(proposalId, GetYesVotes(proposalId) + 1);
+        }
+        else
+        {
+            SetNoVotes(proposalId, GetNoVotes(proposalId) + 1);
+        }
+    }
+
+    private void Unvote(uint proposalId, Votes currentVote)
+    {
+        switch (currentVote)
+        {
+            case Votes.Yes: SetYesVotes(proposalId, GetYesVotes(proposalId) - 1); break;
+            case Votes.No: SetNoVotes(proposalId, GetNoVotes(proposalId) - 1); break;
+            case Votes.None: break;
+        }
+    }
+
+    private Votes ToVote(bool vote) => vote ? Votes.Yes : Votes.No;
+
+    public void ExecuteProposal(uint proposalId)
+    {
+        var proposal = GetProposal(proposalId);
+
+        Assert(proposal.Owner == Message.Sender, "The proposal can be executed by proposal creator.");
+
+        var yesVotes = GetYesVotes(proposalId);
+        var noVotes = GetNoVotes(proposalId);
+
+        Assert(yesVotes > noVotes, "The proposal voting is not passed.");
+        Assert(yesVotes >= MinQuorum, "Min quorum for proposal is not reached.");
+
+        Assert(!proposal.Executed, "The proposal is already executed.");
+        Assert(GetVotingDeadline(proposalId) < Block.Number, "Voting is still open for the proposal.");
+        Assert(proposal.RequestedAmount <= Balance, "Insufficient balance.");
+
+        proposal.Executed = true;
+
+        SetProposal(proposalId, proposal);
+        var result = Transfer(proposal.Recipient, proposal.RequestedAmount);
+
+        Assert(result.Success, "Transfer failed.");
+
+    }
+
+    public void BlacklistAddress(Address address)
+    {
+        EnsureOwnerOnly();
+        SetIsWhitelisted(address, false);
+    }
+
+    public void BlacklistAddresses(byte[] addresses)
+    {
+        EnsureOwnerOnly();
+        foreach (var address in Serializer.ToArray<Address>(addresses))
+        {
+            SetIsWhitelisted(address, false);
+        }
+    }
+
+    public void WhitelistAddress(Address address)
+    {
+        EnsureOwnerOnly();
+        SetIsWhitelisted(address, true);
+    }
+
+    public void WhitelistAddresses(byte[] addresses)
+    {
+        EnsureOwnerOnly();
+        foreach (var address in Serializer.ToArray<Address>(addresses))
+        {
+            SetIsWhitelisted(address, true);
+        }
+    }
+
+    private void EnsureOwnerOnly() => Assert(this.Owner == Message.Sender, "The method is owner only.");
+
+    /// <summary>
+    /// Public method for allow deposits. 
+    /// </summary>
+    public void Deposit()
+    {
+
```

Comment:
I think a log here would be nice, would be good to have an easy to see history of who deposited and how much